model/project.go

```diff
@@ -1128,6 +1128,8 @@ func (p *Project) FindTaskForVariant(task, variant string) *BuildVariantTaskUnit
 			if projectTask := p.FindProjectTask(task); projectTask != nil {
 				bvt.Populate(*projectTask)
 				return &bvt
+			} else if _, exists := tgMap[task]; exists {
```

Comment:
This might seem like a really random change, but it's necessary to make selecting task groups as a dependency work (it's tested by https://github.com/evergreen-ci/evergreen/blob/d42bad06c9835faccf57278158f662ceac0de7f0/model/patch_lifecycle_test.go#L539). In this pr, this function gets passed a variant and either a task name or task group name. If it's a task name things work already. If it's a task group, we don't currently return anything here, even though you can add an entire task group to a variant

Reply:
Oh okay I see, thanks for the comment